Fix the lint issues

ramya-bangera opened the pull request:
1. fix errcheck in main.go - check return value of `viper.BindPFlags`
2. fix v in config.go - added nolint to config.go to escape from lint issues as it contains unused variables. This file can even be removed from repo if not needed

dhui committed:
Use newer math/rand/v2

dhui committed:
Fix redshift tests

Newer versions of postgres require a password

dhui committed:
Use forked version of postgres 8 to avoid Docker image format version issues.
See: https://docs.docker.com/engine/deprecated/#pushing-and-pulling-with-image-manifest-v2-schema-1

Forked images created by pulling the original image and pushing to the fork. For more details, see: https://github.com/golang-migrate/migrate/pull/1143#issuecomment-2328069381

kd7lxl commented on cmd/migrate/main.go:
```diff
@@ -16,7 +16,7 @@ import (
 func init() {
 	pflag.Parse()
-	viper.BindPFlags(pflag.CommandLine)
+	_ = viper.BindPFlags(pflag.CommandLine)
```
Wouldn't it be better to check this return value than to ignore it?

ramya-bangera replied:
Yes, fixed it as discussed

ramya-bangera committed:
Fix the lint issues

1. fix errcheck in main.go - check return value of `viper.BindPFlags`
2. fix v in config.go - added nolint to config.go to escape from lint issues as it contains unused variables. This file can even be removed from repo if not needed

ramya-bangera commented:
Closing as this requires a full merge from upstream
